code/modules/clothing/masks/miscellaneous.dm

```diff
@@ -52,34 +52,69 @@
 
 //scarves (fit in in mask slot)
 
-/obj/item/clothing/mask/bluescarf
-	name = "blue neck scarf"
-	desc = "A blue neck scarf."
-	icon_state = "blueneckscarf"
-	item_state = "blueneckscarf"
+/obj/item/clothing/mask/scarf
+	name = "scarf"
+	desc = "A simple neck scarf."
+	icon_state = "bluescarf"
+	item_state = "bluescarf"
 	flags = MASKCOVERSMOUTH
 	w_class = 2
 	gas_transfer_coefficient = 0.90
+	action_button_name = "Adjust scarf"
+	var/hanging = 0
+
+/obj/item/clothing/mask/scarf/blue
+	name = "blue neck scarf"
+	desc = "A blue neck scarf."
+	icon_state = "bluescarf"
+	item_state = "bluescarf"
 
-/obj/item/clothing/mask/redscarf
+/obj/item/clothing/mask/scarf/red
 	name = "red scarf"
-	desc = "A red and white checkered neck scarf."
-	icon_state = "redwhite_scarf"
-	item_state = "redwhite_scarf"
-	flags = MASKCOVERSMOUTH
-	w_class = 2
-	gas_transfer_coefficient = 0.90
+	desc = "A red neck scarf."
+	icon_state = "redscarf"
+	item_state = "redscarf"
 
-/obj/item/clothing/mask/greenscarf
+/obj/item/clothing/mask/scarf/green
 	name = "green scarf"
 	desc = "A green neck scarf."
-	icon_state = "green_scarf"
-	item_state = "green_scarf"
-	flags = MASKCOVERSMOUTH
-	w_class = 2
-	gas_transfer_coefficient = 0.90
+	icon_state = "greenscarf"
+	item_state = "greenscarf"
+
+/obj/item/clothing/mask/scarf/yellow
+	name = "yellow scarf"
+	desc = "A yellow neck scarf."
+	icon_state = "yellowscarf"
+	item_state = "yellowscarf"
+
+/obj/item/clothing/mask/scarf/violet
+	name = "violet scarf"
+	desc = "A violet neck scarf."
+	icon_state = "violetscarf"
+	item_state = "violetscarf"
+
+/obj/item/clothing/mask/scarf/attack_self()
+
+	if(usr.canmove && !usr.stat && !usr.restrained())
+		if(!src.hanging)

```

Comment:
https://github.com/TauCetiStation/TauCetiClassic/wiki/Code-Convention#%D0%98%D1%81%D0%BF%D0%BE%D0%BB%D1%8C%D0%B7%D0%BE%D0%B2%D0%B0%D0%BD%D0%B8%D0%B5-src